src/loan_form/LoanForm.jsx

```diff
@@ -22,82 +21,67 @@ export default class LoanForm extends React.Component {
   }
   
   @autobind
+  handleAddLoan() {
+    const newLoans = this.state.loans.concat(
+      this.createLoan()
+    )
+
+    this.setState({ loans: newLoans })
+  }
+  
+  @autobind
   handleChange(id, name, value) {
     const newLoans = this.state.loans
 
+    const dashCaseToCamelCase = string => {
+      return string.replace(/-[a-z]/, match => {
+        return match.substring(1).toUpperCase()
+      })
+    }
+
     for (var i = 0; i < this.state.loans.length; i++) {
       if (this.state.loans[i]['id'] == parseInt(id)) {
-        switch(name) {
-          case 'title':
-            newLoans[i].title = value
-            break
-          case 'amount-owed':
-            newLoans[i].amountOwed = value
-            newLoans[i].totalMonths = this.findLoanLength(parseFloat(newLoans[i].amountOwed),
-                                                          parseFloat(newLoans[i].interestRate),
-                                                          parseFloat(newLoans[i].monthlyPayment))
-            console.log(newLoans[i])
-            break
-          case 'interest-rate':
-            newLoans[i].interestRate = value
-            newLoans[i].totalMonths = this.findLoanLength(parseFloat(newLoans[i].amountOwed),
-                                                          parseFloat(newLoans[i].interestRate),
-                                                          parseFloat(newLoans[i].monthlyPayment))
-            break
-          case 'monthly-payment':
-            newLoans[i].monthlyPayment = value
-            newLoans[i].totalMonths = this.findLoanLength(parseFloat(newLoans[i].amountOwed),
-                                                          parseFloat(newLoans[i].interestRate),
-                                                          parseFloat(newLoans[i].monthlyPayment))
-            break
-        }
+        newLoans[i][dashCaseToCamelCase(name)] = value
+        newLoans[i].monthsToPayOffLoan = this.findLoanLength(newLoans[i])
       }
     }
-    console.log(newLoans)
-    this.setState({ loans: newLoans })
-    console.log(this.state)
-  }
-
-  @autobind
-  handleAddLoan() {
-    const newLoans = this.state.loans.concat(
-      this.createLoan()
-    )
 
     this.setState({ loans: newLoans })
   }
 
   @autobind
-  findLoanLength(amountOwed, interestRate, monthlyPayment) {
-    let tempAmountOwed = amountOwed
-    let monthlyInterest = tempAmountOwed * ((interestRate / 365) ** 30.417)
-    let totalMonths = 0
+  findLoanLength({amountOwed, interestRate, monthlyPayment}) {
+    const daysInYear = (year) => 365
+    const daysInMonth = (month) => 30.417
+    const effectiveMonthlyInterest = (amountOwed, yearlyInterestRate, month, year) => {
+      return amountOwed * (((1 + yearlyInterestRate / daysInYear(year)) ** daysInMonth(month)) - 1)
+    }
```

Comment:
lol when i said we should lift these into functions, i meant that they should be stand-alone or member methods of the LoanForm class, not named consts in the body of this method :)